src/dmd/attrib.d

```diff
@@ -54,20 +54,7 @@ extern (C++) abstract class AttribDeclaration : Dsymbol
 
     override final int apply(Dsymbol_apply_ft_t fp, void* param)
     {
-        Dsymbols* d = include(_scope);
-        if (d)
-        {
-            for (size_t i = 0; i < d.dim; i++)
-            {
-                Dsymbol s = (*d)[i];
-                if (s)
-                {
-                    if (s.apply(fp, param))
-                        return 1;
-                }
-            }
-        }
-        return 0;
+        return include(_scope).foreachDsymbol( (s) { return s && s.apply(fp, param); } );
```

Comment:
Use:

```suggestion
        return include(_scope).foreachDsymbol(s => s && s.apply(fp, param));
```

Or at least:

```suggestion
        return include(_scope).foreachDsymbol( s => s && s.apply(fp, param) );
```

If you demand on having whitespace around the lambda

Reply:
Again, the `return` is needed for the early out loop.

Reply:
The two syntaxes have equivalent semantics, as far as I know. You would still break from the loop as soon as `apply` retuens non-zero.